Fix bz67675

markt-asf opened the pull request:
Fixes https://bz.apache.org/bugzilla/show_bug.cgi?id=67675

providing as a PR to allow review before commit

michael-o commented:
There is also AES128 with OpenSSL anf should be tried as well.

michael-o commented on java/org/apache/tomcat/util/net/jsse/PEMFile.java:
```diff
@@ -64,8 +66,40 @@ public class PEMFile {
+    }
 
-    private static final String PBES2 = "PBES2";
+    private static final Map<String,Algorithm> OID_TO_ALGOITHM = new HashMap<>();
```
Typo

michael-o commented:
The more ASN.1 I see in Tomcat the more I think is that we should import Kerby ASN.1 at some point in time...

michael-o commented on test/org/apache/tomcat/util/net/jsse/TestPEMFile.java:
```diff
@@ -118,4 +127,17 @@ private String getPath(String file) throws IOException {
+            System.out.println( provider );
+            for (Provider.Service service : provider.getServices()) {
+                if ("Cipher".equals( service.getType())) {
+                    System.out.println( service.getAlgorithm() );
```
Are the spaces on purpose?

michael-o commented:
Thanks for the hard work. I will try this tomorrow/Wednesday!

michael-o commented:
Built from your branch and I can confirm that this works with a selfsigned and one issued by our enterprise CA.

markt-asf commented:
Added a test key using AES128 (test passed without additional code changes).
Fixed typo.
Removed debug code that wasn't intended to be part of the commit.
I might be persuaded to include Kerby ASN.1 although I'm not a fan of additional dependencies. We'd need to shade it if we did. Something to consider alongside the whether we want to change how we handle the Commons packages.

michael-o commented:
Yes, through shading only to make them private. Looking at the vast amount of code you have written already I bet that ther are also others which would like to use such a PEM to Java Keystore converter. I consider it a pain everytime to have the tripe of key, cert and password and them being forced just for Java to convert it.
